packages/syft/src/syft/custom_worker/k8s.py

```diff
@@ -1,24 +1,36 @@
 # stdlib
+import base64
 from enum import Enum
+from functools import cache
+import json
 import os
+from typing import Dict
+from typing import Iterable
+from typing import List
 from typing import Optional
+from typing import Tuple
+from typing import Union
 
 # third party
-from kr8s._data_utils import list_dict_unpack
+import kr8s
+from kr8s.objects import APIObject
+from kr8s.objects import Pod
+from kr8s.objects import Secret
 from pydantic import BaseModel
 
 # Time after which Job will be deleted
 JOB_COMPLETION_TTL = 60
 
-# Persistent volume claim for storing build output
-BUILD_OUTPUT_PVC = "worker-builds"
-
 # Kubernetes namespace
 KUBERNETES_NAMESPACE = os.getenv("K8S_NAMESPACE", "syft")
 
 # Kubernetes runtime flag
 IN_KUBERNETES = os.getenv("CONTAINER_HOST") == "k8s"
 
+# Internal registry URL
+DEFAULT_INTERNAL_REGISTRY = f"registry.{KUBERNETES_NAMESPACE}.svc.cluster.local"
+INTERNAL_REGISTRY_HOST = os.getenv("INTERNAL_REGISTRY_HOST", DEFAULT_INTERNAL_REGISTRY)
```

Comment:
Optionally, in dev mode, we can simply set `INTERNAL_REGISTRY_HOST=k3d-registry.localhost:5000`